dma-side-channel/main.c

```diff
@@ -35,28 +32,30 @@ void SM_ENTRY(foo) test(char key) {
 }
 
 /*
- * For the JMP 2f instruction, the program memory trace will be 00011111 10000000
- * For the NOP    instruction, the program memory trace will be 00011111 00000000
+ * For the JMP 2f instruction, the program memory trace will be 11110000 00000000
+ * For the NOP    instruction, the program memory trace will be 11100000 00000000
+ * The JMP is instruction is executed if the guess was incorrect, the NOP if it
+ * was correct.
  *
- * By checking the different bit at the 0x40 position, we can tell which branch
- * was taken (the JPM is executed if the guess was incorrect, after the
- * bit is set).
+ * Using TRACE_MASK to get the bit that differs in the two possible traces, we
+ * can tell whether a collected trace was generated as a result of a correct or
+ *  an incorrect guess.
  */
 
 void irqHandler(void)
 {
-    delay = __ss_isr_reti_latency + DELAY_BEFORE_SM;
+    __ss_set_dma_attacker_delay(0);
     uint16_t *pmem_addr = (uint16_t*) irqHandler;
     if (instruction_counter == INSTRUCTION_NUMBER_JMP) {
-        if (*dma_trace & 0x80) {
+        if (DMA_TRACE & TRACE_MASK) {
             pr_info("Key was not guessed!");
         } else {
             pr_info("Key was guessed!");
         }
     }
     ++instruction_counter;
-    *dma_addr = pmem_addr;
-    *dma_countdown = delay;
+    DMA_ADDR = pmem_addr;
+    DMA_COUNTDOWN = delay;
```

Comment:
why is this still necessary? Is the idea not to do this all transparently in the sancus-step asm framework code?

Reply:
You're right, setting the delay from here actually has no effect, this was left over from the previous implementation.